### PR TITLE
support types_or

### LINKED ISSUE #69: Add support for `types_or`
Looks like `--types` is supported by not `types_or` equivalent.

### PR DESCRIPTION
**Pull request by henryiii**
Closes #69.

### CODE REVIEW

**Comment by henryiii** [comment]
Not related to this pr, but shouldn't flake8 be below some of the other hooks, like autopep8? autopep8 is a modifying hook, and flake8 is not.

File: tox.ini [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = py36,py37,pypy3,pre-commit
+envlist = py36,py39,pypy3,pre-commit
```
Comment (henryiii) [inline tox.ini:2]: Oops, sorry, didn't mean to leave this in. I don't have 3.6 or 3.7 locally.

Comment (henryiii) [reply tox.ini]: And I didn't realize `tox -e py39` would work without this change, but it seems to.

**Comment by henryiii** [comment]
(totally forgot about the minimum required even after asking about it, will add then un-draft)

**Comment by henryiii** [comment]
Are you okay with adding `minimum_pre_commit_version: '0'` to the output on non-type_or repos, or should that line be conditional, only if non zero?

**Comment by henryiii** [comment]
@asottile, I assumed you were fine with `minimum_pre_commit_version: '0'` showing up in all non-types_or repos, but happy to change that if it is better to make it conditional, only showing up if set. I think this is ready for review.

File: pre_commit_mirror_maker/main.py [hunk 2]
```diff
@@ -70,16 +71,30 @@ def main(argv: Optional[Sequence[str]] = None) -> int:
     )
     args = parser.parse_args(argv)
 
+    minimum_pre_commit_version = '0'
+
+    if args.types_or:
+        match_key = 'types_or'
+        match_val = f'[{args.types_or}]'
```
Comment (asottile) [inline pre_commit_mirror_maker/main.py:78]: unlike `types`, `types_or` is usually multiple things

File: pre_commit_mirror_maker/main.py [hunk 3]
```diff
@@ -70,16 +71,30 @@ def main(argv: Optional[Sequence[str]] = None) -> int:
...
+        minimum_pre_commit_version = '2.9.0'
```
Comment (asottile) [inline pre_commit_mirror_maker/main.py:79]: 2.9.2 is a slightly better version, `types_or` was subtly broken in the initial release

File: tests/main_test.py [hunk 4]
```diff
@@ -75,3 +75,40 @@ def test_main_with_args(mock_make_repo):
     ))
     expected = '["-i", "--ignore=E265,E501"]'
     assert mock_make_repo.call_args[1]['args'] == expected
+
+
+def test_main_with_types_and_types_or(mock_make_repo):
+    with pytest.raises(SystemExit):
```
Comment (asottile) [inline tests/main_test.py:81]: this should specifically check for a failure and/or a message

File: pre_commit_mirror_maker/main.py (hunk 2 above)
Comment (henryiii) [reply pre_commit_mirror_maker/main.py:78]: How would you like to enter it? I assumed `--types-or=a,b,c` similar to `--types=text,py`. An argparse list?

Comment (asottile) [reply pre_commit_mirror_maker/main.py:78]: action='append' probably

**Comment by henryiii** [comment]
Manually writing `--types-or=a,b` still works, but now `--types-or=a --types-or=b` also works. I did not apply this to `--types` (but I could if you want). I think everything is addressed.

**Review by asottile — APPROVED** [review]

**Pull request merged**